This changes allow to disable the pll algorithm to change the timestamp of the message header

cmuehlbacher opened the pull request:
This should be reviewd after the merge of the #1 .
It adds a prameter use_pll_correction to define if the pll algorithm is used to adapt the timestamp within the message header.
If set to false the timestamp of the scanner is used. This reduces jitter for new laser scan (e.g. the pico scan)

cmuehlbacher committed:
add paramter to disable softwarepll if desired

cmuehlbacher committed:
fixed bug causing the pll correction paramter not be used

cmuehlbacher committed:
fixed bug causing the pll not to be properly disabled

cmuehlbacher committed:
add documentation about the parameter to enable/disable the pll algorithm

wedenikf commented on driver/src/softwarePLL.cpp:
```diff
@@ -184,14 +198,14 @@ bool SoftwarePLL::updatePLL(uint32_t sec, uint32_t nanoSec, uint32_t curtick)
 //TODO Kommentare
 bool SoftwarePLL::getCorrectedTimeStamp(uint32_t &sec, uint32_t &nanoSec, uint32_t curtick)
 {
-  if (IsInitialized() == false)
+  if ((IsInitialized() == false) && (correctionDisabled == false))
   {
     return (false);
   }
```
I don't know how this method is used, but your condition seems odd to me.

What should happen for `IsInitialized() == true` and `correctionDisabled == true`? I guess we want to return `false` in case we don't correct the time stamp. With both variables as `true`, the condition evaluates to `false` and we would modify the time stamp. That means that `correctionDisabled` doesn't fulfill its job here, right?

Please add another variable here that gives the condition a meaningful name.

wedenikf reviewed (changes requested):
I only focused on the commits that are not part of #1. The PR still shows them as "wants to merge", can you please check what's up with that?

cmuehlbacher committed:
Merge branch 'master' into feature/disable_pll

cmuehlbacher committed:
clarified get corrected time stamp

ensured that everything used is initialized
split case of correction enabled/disabled to ease maintenance
add comments about the function and its usage

wedenikf reviewed (approved):
LGTM
